singularity-eos/eos/eos.hpp

```diff
@@ -125,10 +125,19 @@ class Gruneisen : public EosBase<Gruneisen> {
  public:
   Gruneisen() = default;
   PORTABLE_INLINE_FUNCTION
+  Gruneisen(const Real C0, const Real s1, const Real s2, const Real s3, const Real G0,
+            const Real b, const Real rho0, const Real T0, const Real P0, const Real Cv,
+            const Real rho_max)
+      : _C0(C0), _s1(s1), _s2(s2), _s3(s3), _G0(G0), _b(b), _rho0(rho0), _T0(T0), _P0(P0),
+        _Cv(Cv), _rho_max(rho_max) {}
```

Comment:
Should there be a check here in the constructor to ensure that user-specified rho_max is sane (and, for instance, less than the result of `ComputeRhoMax`)?

Reply:
I would prefer to warn rather than die if rho_max is specified to be greather than `computeRhoMax`  but a warning would be good maybe.

Reply:
I'll create a hotfix for this shortly.

Just to be sure, @Yurlungur , we don't have a full-fledged warning mechanism other than print statements, right?

Reply:
We don't. Here's my suggested design for now:
```C++
#ifndef NDEBUG
if (rho max condition violated) {
printf("some warning message);
}
#endif
```

I'll try to submit some more formal warning machinery as a PR at some point

Reply:
Updated here:
https://github.com/lanl/singularity-eos/pull/130